pkg/acsengine/defaults.go

```diff
@@ -587,6 +587,11 @@ func setOrchestratorDefaults(cs *api.ContainerService) {
 		if a.MasterProfile.Distro == "" {
 			a.MasterProfile.Distro = api.RHEL
 		}
+		for key, agent := range a.AgentPoolProfiles {
```

Comment:
strangely, it looks like the default to ubuntu happens in `setMasterNetworkDefaults` and `setAgentNetworkDefaults`.  That seems incorrect to me.  I like it here.  Just wanted to point out that if we commit here this logic is now separated for better or worse.

In any case....I'm playing unit test police for everything.  Where them tests at? (looks like it will be the first one for this case, you only have to cover what you added)  :smile: :police_car: :oncoming_police_car:

Reply:
@pweil- are you saying that `set[components]NetworkDefaults` should not be tempering with `Distro` in the first place? And we should just move all `Distro` logic to `setOrchestratorDefaults` or all orchestrators? (assuming I will write the testcase first and it will backward compatible :blush:

Reply:
it shouldn't, but until there's a better interface around the orchestrators I'd personally be tempted to condense all the `.Distro = api.RHEL` code next to the ubuntu equivalents (@pweil- don't look at me like that) and write a unit test.

Reply:
> don't look at me like that

+1 for consistency,  -0.5 for location.  Overall we're still ahead, go for it.